packages/fetchai/connections/soef/connection.py

```diff
@@ -879,7 +879,13 @@ async def _set_personality_piece_handler(
         if not (isinstance(piece, str) and isinstance(value, str)):  # pragma: nocover
             raise SOEFException.debug("Personality piece bad values provided.")
 
-        await self._set_personality_piece(piece, value)
+        pieces = piece.split(",")
+        values = value.split(",")
+        if len(pieces) != len(values):
+            raise SOEFException.debug("Mismatched number of personality pieces and values.")
+
+        for idx in range(len(pieces)):
+            await self._set_personality_piece(pieces[idx], values[idx])
```

Comment:
Have you checked whether it is possible to set a whole (comma separated) list at once?

Reply:
I have to see the SOEF api for that. Is there such a thing accessible?

Reply:
I had a look at a rough documentation.md file in the soef repo and the command doesn't seem to be getting multiple values (I may be mistaken as there was only some rough explanation there and nothing formal).